### PR TITLE
[dv] Fix VCS compile error

### PR DESCRIPTION
**Pull request by weicaiyang**
Change build opt CFLAGS from `--std=c99 to `--std=c++11`

Signed-off-by: Weicai Yang <weicai@google.com>

### CODE REVIEW

File: hw/dv/tools/dvsim/vcs.hjson [hunk 1]
```diff
@@ -31,7 +31,7 @@
                // otherwise. The double-escaped quotes are because this needs to go inside a string
                // that gets passed to Make (stripping one level of quotes), and then needs to
                // result in an argument with an embedded space (the other one).
-               "-CFLAGS \\\"--std=c99 -fno-extended-identifiers\\\"",
+               "-CFLAGS \\\"--std=c++11 -fno-extended-identifiers\\\"",
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/vcs.hjson:34]: What about Xcelium?

**Review by sriyerg — APPROVED** [review]
LGTM!

**Comment by weicaiyang** [comment]
@rswarbrick 

Looks like the syntax in [dpi_memutil.cc](https://github.com/lowRISC/opentitan/blob/master/hw/dv/verilator/cpp/dpi_memutil.cc) requires `--std=c++11`, otherwise, we will see compile error.

File: hw/dv/tools/dvsim/vcs.hjson (hunk 1 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/vcs.hjson:34]: haven't used xcelium in top-level for a while. probably need to make some separate fixes for it

File: hw/dv/tools/dvsim/vcs.hjson [hunk 2]
```diff
@@ -31,7 +31,9 @@
...
+               // This is required for dpi_memutil.cc
+               "-CFLAGS --std=c++11",
```
Comment (sriyerg) [inline hw/dv/tools/dvsim/vcs.hjson:36]: Probably replace the comment above with something like 'C++11 standard is enforced for all sources written in C++, including the DPI-C constructs.'

File: hw/dv/tools/dvsim/vcs.hjson (hunk 1 above)
Comment (sriyerg) [reply hw/dv/tools/dvsim/vcs.hjson:34]: I was wondering if we should proactively add the same switch to Xcelium as well, but its fine, not a priority.

**Review by rswarbrick — APPROVED** [review]

File: hw/dv/tools/dvsim/vcs.hjson (hunk 1 above)
Comment (weicaiyang) [reply hw/dv/tools/dvsim/vcs.hjson:34]: Xcelium option may be different from VCS. I created an issue #4230 for this.

**Comment by imphil** [comment]
Why not go for c++14 right away, since that's what we have as system requirement?

**Comment by weicaiyang** [comment]
> Why not go for c++14 right away, since that's what we have as system requirement?

@imphil 
In order to use c++14, we need to extra work to make it work in LSF in Google. Our internal tool support suggests to limit to c++11 for VCS. Want to double check that's not the right direction we should go, right?

**Comment by sriyerg** [comment]
Getting this update merged is urgent (outage at Google preventing us from running chip level sims). We can update the C++ standard to '14 only after GCC5 (or newer) is installed within Google compute infra.

**Pull request merged**